### ISSUE TITLE
Merge `RequestDto` with `UpdateDto`

### ISSUE THREAD

**Issue by ndr-brt** [eclipse-edc/Connector#3060]
# Feature Request

Currently we have, for the `Asset`, `ContractDefinition` and `PolicyDefinition` entities, different `Dto`s for "creation" and "update", these could be merged into a single `Dto` each, and the `DtoWrapper` and related transformer will be useless.

To be done after #3054 

## Which Areas Would Be Affected?
api

## Why Is the Feature Desired?
cleanup

## Solution Proposal
_If possible, provide a (brief!) solution proposal._

**Comment by ndr-brt** [comment]
Superseded by #3267
